### ISSUE TITLE
Build fails when using spot_tags

### ISSUE THREAD

**Issue by pcanham** [hashicorp/packer#7941]
When i try and add in spot_tags information into my packer template it errors with the following error, but validation checks on the template pass OK

```
Error tagging spot request: InvalidParameterValue: Value ( null ) for parameter resourceId is invalid. Null/empty value for resourceId is invalid
```

- Packer version from Packer v1.4.2
- Host platform: MacOS 10.14.6 & Amazon Linux 2
- Debug log output from `PACKER_LOG=1 packer build template.json`.
https://gist.github.com/pcanham/6cc1a066ce05fd6166b12cd3ade69245

- The _simplest example template and scripts_ needed to reproduce the bug.
https://gist.github.com/pcanham/9157b72ebf35d1f1fbf5051f5e208dd6

**Comment by SwampDragons** [comment]
It turns out this is a nightmare to untangle because after moving to the "instant" fleet implementation I can't find a way to determine the resource ID for the spot request. I won't be able to fix the tagging before the 1.4.3 release but I will add a warning to explain to users what's going on and that they need to downgrade in the meantime.

**Comment by SwampDragons** [comment]
I accidentally pushed to master, like a monster, but this should now be fixed in 1.4.4-dev
